cmd/main.go

```diff
@@ -1,5 +1,104 @@
 package main
 
+import (
+	"Groupchat-Service/internal/config"
+	"Groupchat-Service/internal/controllers"
+	"Groupchat-Service/internal/database/repositories"
+	"Groupchat-Service/internal/middleware"
+	"Groupchat-Service/internal/services"
+	"context"
+	firebase "firebase.google.com/go/v4"
+	"fmt"
+	"github.com/gin-contrib/cors"
+	"github.com/gin-gonic/gin"
+	_ "github.com/lib/pq"
+	"google.golang.org/api/option"
+	"log"
+	"net/http"
+	"time"
+)
+
 func main() {
+	// Load configuration
+	cfg, err := config.LoadConfig(".")
+	if err != nil {
+		log.Fatalf("Error loading config: %v", err)
+	}
+
+	// Initialize Azure Table client
+	tableClient, err := repositories.NewTableClient(cfg.AzureConnectionString)
+	if err != nil {
+		log.Fatalf("Failed to create table client: %v", err)
+	}
+
+	// Initialize repositories
+	messageRepo, err := repositories.NewMessageRepository(tableClient)
+	if err != nil {
+		log.Fatalf("Failed to create message repository: %v", err)
+	}
+
+	fcmTokenRepo, err := repositories.NewFCMTokenRepository(tableClient)
+	if err != nil {
+		log.Fatalf("Failed to create FCM token repository: %v", err)
+	}
+
+	// Initialize services
+	notificationService, err := services.NewNotificationService(cfg.FirebaseCredentialFile)
+	if err != nil {
+		log.Fatalf("Failed to create notification service: %v", err)
+	}
+
+	messageService := services.NewMessageService(messageRepo, fcmTokenRepo, notificationService)
+	validationService := services.NewValidationService()
+	
+	// Initialize controllers
+	messageController := controllers.NewMessageController(messageService, validationService)
+
+	// Set up router
+	router := gin.Default()
+
+	// Add CORS middleware
+	router.Use(cors.New(cors.Config{
+		AllowOrigins:     []string{"*"},
+		AllowMethods:     []string{"GET", "POST", "PUT", "DELETE"},
+		AllowHeaders:     []string{"Origin", "Content-Type", "Authorization"},
+		ExposeHeaders:    []string{"Content-Length"},
+		AllowCredentials: true,
+		MaxAge:           12 * time.Hour,
+	}))
+
+	// Add Prometheus middleware BEFORE other middleware
+	router.Use(middleware.PrometheusMiddleware())
+
+	// Define routes
+	router.GET("/groups/:groupId/messages", messageController.GetMessages)
+	router.POST("/groups/:groupId/messages", messageController.CreateMessage)
+	router.PUT("/groups/:groupId/messages/:messageId/pin", messageController.ToggleMessagePin)
```

Comment:
Misschien beter om de routes in de controller zelf te configuren met een routes method (zie de authentication service)